File: CS/Models/OrdersModel.cs
using System;

namespace Example.Models {
    public class OrdersModel {

        private readonly NorthwindTableAdapters.OrdersTableAdapter _data = new NorthwindTableAdapters.OrdersTableAdapter();

        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public Northwind.OrdersDataTable Data { get { return _data.GetDataByDate(StartDate, EndDate); } }

        public DateTime MinDate {
            get {
                return (DateTime?)_data.MinOrderDate() ?? new DateTime (1990, 1, 1);
            }
        }

        public DateTime MaxDate {
            get {
                return (DateTime?)_data.MaxOrderDate() ?? new DateTime(2011, 1, 1);
            }
        }
    }
}

File: VB/Models/OrdersModel.vb
Imports Microsoft.VisualBasic
Imports System
Imports Example.NorthwindTableAdapters

Public Class OrdersModel

    Private ReadOnly _data As OrdersTableAdapter = New OrdersTableAdapter()

    Private privateStartDate As DateTime
    Public Property StartDate() As DateTime
        Get
            Return privateStartDate
        End Get
        Set(ByVal value As DateTime)
            privateStartDate = value
        End Set
    End Property
    Private privateEndDate As DateTime
    Public Property EndDate() As DateTime
        Get
            Return privateEndDate
        End Get
        Set(ByVal value As DateTime)
            privateEndDate = value
        End Set
    End Property
    Public ReadOnly Property Data() As Northwind.OrdersDataTable
        Get
            Return _data.GetDataByDate(StartDate, EndDate)
        End Get
    End Property

    Public ReadOnly Property MinDate() As DateTime
        Get
            Return If(CType(_data.MinOrderDate(), Nullable(Of DateTime)).HasValue, CType(_data.MinOrderDate(), Nullable(Of DateTime)).Value, New DateTime(1990, 1, 1))
        End Get
    End Property

    Public ReadOnly Property MaxDate() As DateTime
        Get
            Return If(CType(_data.MaxOrderDate(), Nullable(Of DateTime)).HasValue, CType(_data.MaxOrderDate(), Nullable(Of DateTime)).Value, New DateTime(2011, 1, 1))
        End Get
    End Property
End Class
